src/PlanOnlyStrategy.groovy

```diff
@@ -0,0 +1,37 @@
+import static TerraformEnvironmentStage.ALL
+import static TerraformEnvironmentStage.PLAN
+
+class PlanOnlyStrategy {
+
```

Comment:
This is making me really uncomfortable from a maintainability perspective.  I've already been chewing on removing the Strategy implementation from the 5.9 release for this reason.  I think more work needs to go into the Strategy idea before we start building plugins on it.